internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -248,7 +255,92 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 			return fmt.Errorf("error creating address spec: %w", err)
 		}
 
-		// all done, terminate controller
-		return nil
+		return ctrl.pollConnection(ctx, logger)
```

Comment:
wouldn't this re-create the `wgclient` every time, should we cache it?

Reply:
This polling is kinda long-lived: as long as the connection stays alive, it'll keep using the same `wgClient`. I'm not sure if further caching is necessary tbh.

Reply:
This changed a bit as well - now the client lives as long as the controller does not crash - so it is even longer lived now.